### PR TITLE
[1-0004: Create REST to get list of all restaurants] 

### PR DESCRIPTION
**Pull request by Saparmyrat12**

### CODE REVIEW

File: epam-vote-system-controller/src/main/java/com/epam/vote/controller/RestaurantController.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+/**
+ * This class handles requests from the client.
```
Comment (AzarenkaA) [inline epam-vote-system-controller/src/main/java/com/epam/vote/controller/RestaurantController.java:23]: Wrong comment format

File: epam-vote-system-controller/src/main/java/com/epam/vote/controller/RestaurantController.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
+/**
+ * <p/>
+ * Copyright (C) 2021
+ * <p/>
+ * Date: май 24, 2021
+ * @author Sapar
+ */
```
Comment (AzarenkaA) [inline epam-vote-system-controller/src/main/java/com/epam/vote/controller/RestaurantController.java:7]: What is it? Move it to right place

File: epam-vote-system-controller/src/main/java/com/epam/vote/controller/RestaurantController.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+@RestController
+@RequestMapping("/api/restaurant")
+public class RestaurantController {
+
+    @Autowired
+    private RestaurantService restaurantService;
+
+    @GetMapping
+    public ResponseEntity<List<RestaurantDto>> get() {
+        List<RestaurantDto> entities = StreamSupport
```
Comment (AzarenkaA) [inline epam-vote-system-controller/src/main/java/com/epam/vote/controller/RestaurantController.java:34]: Why do you use StreamSupport? Use stream

File: epam-vote-system-domain/src/main/java/com/epam/vote/domain/Restaurant.java [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+ * <p/>
+ * Date: май 19, 2021
+ * @author Sapar
+ */
+package com.epam.vote.domain;
+
+import java.time.LocalDate;
+
+/**
+ * This class describe table Restaurant.
```
Comment (AzarenkaA) [inline epam-vote-system-domain/src/main/java/com/epam/vote/domain/Restaurant.java:13]: The same, fix comment in thes class

File: epam-vote-system-domain/src/main/java/com/epam/vote/domain/Restaurant.java [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+    }
+
+    public int getRecordVersion() {
+        return recordVersion;
+    }
+
+    public void setRecordVersion(int recordVersion) {
+        this.recordVersion = recordVersion;
+    }
+
```
Comment (AzarenkaA) [inline epam-vote-system-domain/src/main/java/com/epam/vote/domain/Restaurant.java:104]: Empty lines

File: epam-vote-system-controller/src/main/java/com/epam/vote/controller/RestaurantController.java [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+                .stream(restaurantService.selectAllRestaurant().spliterator(), false)
+                .map(RestaurantDto::new).collect(Collectors.toList());
+        if (entities.isEmpty()) {
```
Comment (AzarenkaA) [inline epam-vote-system-controller/src/main/java/com/epam/vote/controller/RestaurantController.java:37]: don't use if statment. replace it on ternary operation

File: epam-vote-system-controller/src/main/java/com/epam/vote/controller/RestaurantController.java [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+            return new ResponseEntity<>(HttpStatus.NOT_FOUND);
+        }
+        return new ResponseEntity<>(entities, HttpStatus.OK);
```
Comment (AzarenkaA) [inline epam-vote-system-controller/src/main/java/com/epam/vote/controller/RestaurantController.java:40]: Tests?

File: epam-vote-system-domain/src/main/java/com/epam/vote/domain/Restaurant.java (hunk 5 above)
Comment (AzarenkaA) [reply epam-vote-system-domain/src/main/java/com/epam/vote/domain/Restaurant.java:104]: Whehe equals and hashcode? and to string?

File: epam-vote-system-domain/src/main/java/com/epam/vote/domain/dto/RestaurantDto.java [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+        this.name = name;
+    }
+
+    public String getAddress() {
+        return address;
+    }
+
+    public void setAddress(String address) {
+        this.address = address;
+    }
```
Comment (AzarenkaA) [inline epam-vote-system-domain/src/main/java/com/epam/vote/domain/dto/RestaurantDto.java:48]: The same equals.... etc.

File: epam-vote-system-repository/src/main/java/com/epam/vote/repository/IRestaurantDao.java [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+import com.epam.vote.domain.Restaurant;
+import org.springframework.stereotype.Repository;
+
+import java.util.List;
+
+/**
+ * This class describe method for Restaurant.
+ */
+@Repository
+public interface IRestaurantDao {
```
Comment (AzarenkaA) [inline epam-vote-system-repository/src/main/java/com/epam/vote/repository/IRestaurantDao.java:20]: Rename this class as Repository. IRestaurantRepository

File: epam-vote-system-repository/src/main/java/com/epam/vote/repository/IRestaurantDao.java [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+ */
+package com.epam.vote.repository;
+
+import com.epam.vote.domain.Restaurant;
+import org.springframework.stereotype.Repository;
+
+import java.util.List;
+
+/**
+ * This class describe method for Restaurant.
```
Comment (AzarenkaA) [inline epam-vote-system-repository/src/main/java/com/epam/vote/repository/IRestaurantDao.java:17]: Comment...

File: epam-vote-system-repository/src/main/resources/com.epam.vote.system.repository.mapper/RestaurantMapper.xml [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+        <result column="updatedUser" jdbcType="VARCHAR" property="updated_user"/>
+        <result column="createdDate" jdbcType="TIMESTAMP" property="created_date"/>
+        <result column="updatedDate" jdbcType="TIMESTAMP" property="updated_date"/>
+        <result column="recordVersion" jdbcType="INTEGER" property="record_version"/>
+    </resultMap>
+
+    <select id="selectAllRestaurant" resultMap="RestaurantResultMap">
+        select * from epam.restaurant
+    </select>
+</mapper>
```
Comment (AzarenkaA) [inline epam-vote-system-repository/src/main/resources/com.epam.vote.system.repository.mapper/RestaurantMapper.xml:20]: add empty line

File: epam-vote-system-repository/src/main/resources/com.epam.vote.system.repository.mapper/RestaurantMapper.xml [hunk 12]
```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<!DOCTYPE mapper PUBLIC "-//mybatis.org//DTD Mapper 3.0//EN"
+        "http://mybatis.org/dtd/mybatis-3-mapper.dtd">
+
+<mapper namespace="com.epam.vote.repository.IRestaurantDao">
+    <resultMap id="RestaurantResultMap" type="com.epam.vote.domain.Restaurant">
+        <result column="id" jdbcType="VARCHAR" property="id"/>
```
Comment (AzarenkaA) [inline epam-vote-system-repository/src/main/resources/com.epam.vote.system.repository.mapper/RestaurantMapper.xml:7]: remove jdbcType

File: epam-vote-system-repository/src/main/java/com/epam/vote/repository/IRestaurantDao.java [hunk 13]
```diff
@@ -0,0 +1,22 @@
...
+ */
+@Repository
+public interface IRestaurantDao {
+    List<Restaurant> selectAllRestaurant();
```
Comment (AzarenkaA) [inline epam-vote-system-repository/src/main/java/com/epam/vote/repository/IRestaurantDao.java:21]: lets use prefix find in methods naming  in repository layer

File: epam-vote-system-service/src/main/java/com/epam/vote/service/RestaurantService.java [hunk 14]
```diff
@@ -0,0 +1,28 @@
...
+package com.epam.vote.service;
+
+import com.epam.vote.domain.Restaurant;
+import com.epam.vote.repository.IRestaurantDao;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import java.util.List;
+
+/**
+ * This class describe method for controller.
```
Comment (AzarenkaA) [inline epam-vote-system-service/src/main/java/com/epam/vote/service/RestaurantService.java:17]: The same....

File: epam-vote-system-controller/pom.xml [hunk 15]
```diff
@@ -20,12 +20,6 @@
             <version>${vote.system.version}</version>
         </dependency>
 
-        <dependency>
```
Comment (AzarenkaA) [inline epam-vote-system-controller/pom.xml:23]: What is the reason remove this code?

File: epam-vote-system-service/src/main/java/com/epam/vote/service/RestaurantService.java [hunk 16]
```diff
@@ -0,0 +1,28 @@
...
+ */
+@Service
+public class RestaurantService {
+
+    @Autowired
+    private IRestaurantDao restaurantDao;
+
+    public List<Restaurant> selectAllRestaurant() {
+        return restaurantDao.selectAllRestaurant();
```
Comment (AzarenkaA) [inline epam-vote-system-service/src/main/java/com/epam/vote/service/RestaurantService.java:26]: getAllRestaurants... and where tests?

File: epam-vote-system-domain/src/main/java/com/epam/vote/domain/Restaurant.java [hunk 17]
```diff
@@ -0,0 +1,106 @@
...
+ * @author Sapar
+ */
+package com.epam.vote.domain;
+
+import java.time.LocalDate;
+
+/**
+ * This class describe table Restaurant.
+ */
+public class Restaurant {
```
Comment (AzarenkaA) [inline epam-vote-system-domain/src/main/java/com/epam/vote/domain/Restaurant.java:15]: Use inheritance and abstract class for common fields

File: pom.xml [hunk 18]
```diff
@@ -47,6 +49,14 @@
                     </execution>
                 </executions>
             </plugin>
+            <plugin>
```
Comment (AzarenkaA) [inline pom.xml:52]: why do we need this plugin?

File: epam-vote-system-controller/src/main/resources/com.epam.vote.system.controller/spring/spring-database.xml [hunk 19]
```diff
@@ -21,4 +21,21 @@
         <property name="dataSource" ref="dataSource"/>
         <property name="changeLog" value="${spring.datasource.changeLogFile}"/>
     </bean>
+
+    <bean name="transactionManager"
```
Comment (AzarenkaA) [inline epam-vote-system-controller/src/main/resources/com.epam.vote.system.controller/spring/spring-database.xml:25]: why do you add transaction manager if you dont use it?

File: epam-vote-system-domain/src/main/java/com/epam/vote/domain/dto/RestaurantDto.java [hunk 20]
```diff
@@ -0,0 +1,49 @@
...
+ */
+package com.epam.vote.domain.dto;
+
+import com.epam.vote.domain.Restaurant;
+
+/**
+ * This class is used for dto Restaurant.
+ */
+public class RestaurantDto {
+    private String id;
```
Comment (AzarenkaA) [inline epam-vote-system-domain/src/main/java/com/epam/vote/domain/dto/RestaurantDto.java:16]: Empty line

File: epam-vote-system-repository/pom.xml [hunk 21]
```diff
@@ -14,8 +14,26 @@
     <dependencies>
         <dependency>
             <groupId>com.epam</groupId>
-            <artifactId>epam-vote-system-db</artifactId>
+            <artifactId>epam-vote-system-domain</artifactId>
             <version>${vote.system.version}</version>
         </dependency>
+
+        <dependency>
```
Comment (AzarenkaA) [inline epam-vote-system-repository/pom.xml:21]: why web mvc in repository layer?

File: epam-vote-system-repository/src/main/resources/com.epam.vote.system.repository.mapper/RestaurantMapper.xml [hunk 22]
```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<!DOCTYPE mapper PUBLIC "-//mybatis.org//DTD Mapper 3.0//EN"
+        "http://mybatis.org/dtd/mybatis-3-mapper.dtd">
+
+<mapper namespace="com.epam.vote.repository.IRestaurantDao">
+    <resultMap id="RestaurantResultMap" type="com.epam.vote.domain.Restaurant">
```
Comment (AzarenkaA) [inline epam-vote-system-repository/src/main/resources/com.epam.vote.system.repository.mapper/RestaurantMapper.xml:6]: use alias in mybatis config

File: epam-vote-system-controller/src/main/resources/com.epam.vote.system.controller/spring/mybatis-config.xml [hunk 23]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (AzarenkaA) [inline epam-vote-system-controller/src/main/resources/com.epam.vote.system.controller/spring/mybatis-config.xml:1]: move it in repository layer

File: epam-vote-system-service/src/main/java/com/epam/vote/service/RestaurantService.java [hunk 24]
```diff
@@ -0,0 +1,28 @@
...
+import com.epam.vote.repository.IRestaurantDao;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import java.util.List;
+
+/**
+ * This class describe method for controller.
+ */
+@Service
+public class RestaurantService {
```
Comment (AzarenkaA) [inline epam-vote-system-service/src/main/java/com/epam/vote/service/RestaurantService.java:20]: In service layer we always use interfaces

File: epam-vote-system-controller/src/main/java/com/epam/vote/controller/RestaurantController.java [hunk 25]
```diff
@@ -0,0 +1,42 @@
...
+import org.springframework.web.bind.annotation.RestController;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+/**
+ * This class handles requests from the client.
+ */
+@RestController
+@RequestMapping("/api/restaurant")
```
Comment (AzarenkaA) [inline epam-vote-system-controller/src/main/java/com/epam/vote/controller/RestaurantController.java:26]: URL?

File: epam-vote-system-controller/src/main/java/com/epam/vote/controller/RestaurantController.java [hunk 26]
```diff
@@ -0,0 +1,42 @@
...
+public class RestaurantController {
+
+    @Autowired
+    private RestaurantService restaurantService;
+
+    @GetMapping
```
Comment (AzarenkaA) [inline epam-vote-system-controller/src/main/java/com/epam/vote/controller/RestaurantController.java:32]: it doesn't work

File: epam-vote-system-repository/pom.xml (hunk 21 above)
Comment (Saparmyrat12) [reply epam-vote-system-repository/pom.xml:21]: because I needed to dependencies spring

File: pom.xml (hunk 18 above)
Comment (Saparmyrat12) [reply pom.xml:52]: because I got the error, use java version more than 5

File: epam-vote-system-controller/pom.xml (hunk 15 above)
Comment (Saparmyrat12) [reply epam-vote-system-controller/pom.xml:23]: Becouse it don't use here

File: epam-vote-system-controller/src/main/java/com/epam/vote/controller/RestaurantController.java [hunk 27]
```diff
@@ -0,0 +1,39 @@
...
+@RestController
+@RequestMapping("/restaurant")
+public class RestaurantController {
+
+    @Autowired
+    private RestaurantService restaurantService;
+
+    @GetMapping
+    public ResponseEntity<List<RestaurantDto>> getAllRestaurants() {
+        List<Restaurant> restaurants = restaurantService.getAllRestaurants();
```
Comment (AzarenkaA) [inline epam-vote-system-controller/src/main/java/com/epam/vote/controller/RestaurantController.java:33]: redundand line.

File: epam-vote-system-controller/src/main/java/com/epam/vote/controller/RestaurantController.java (hunk 26 above)
Comment (AzarenkaA) [reply epam-vote-system-controller/src/main/java/com/epam/vote/controller/RestaurantController.java:31]: comment was not resolved

File: epam-vote-system-domain/src/main/java/com/epam/vote/domain/AbstractDomain.java [hunk 28]
```diff
@@ -0,0 +1,48 @@
...
+
+/**
+ * Abstract domain.
+ * <p/>
+ * Copyright (C) 2021
+ * <p/>
+ * Date: май 26, 2021
+ * @author Sapar
+ */
+public abstract class AbstractDomain {
```
Comment (AzarenkaA) [inline epam-vote-system-domain/src/main/java/com/epam/vote/domain/AbstractDomain.java:13]: Rename it as BaseEntity, and make it with generic type. BaseEntity<T>

File: epam-vote-system-domain/src/main/java/com/epam/vote/domain/AbstractDomain.java [hunk 29]
```diff
@@ -0,0 +1,48 @@
...
+
+    private String id;
```
Comment (AzarenkaA) [inline epam-vote-system-domain/src/main/java/com/epam/vote/domain/AbstractDomain.java:15]: what about other common fields?

File: epam-vote-system-service/src/main/java/com/epam/vote/service/IRestaurantService.java [hunk 30]
```diff
@@ -0,0 +1,16 @@
...
+/**
+ * Interface Restaurant service.
+ * <p/>
+ * Copyright (C) 2021
+ * <p/>
+ * Date: май 26, 2021
+ * @author Sapar
+ */
+public interface IRestaurantService {
+    List<Restaurant> getAllRestaurants();
```
Comment (AzarenkaA) [inline epam-vote-system-service/src/main/java/com/epam/vote/service/IRestaurantService.java:15]: Empty line and in interfaces we all make comment

File: epam-vote-system-repository/src/main/java/com/epam/vote/repository/IRestaurantRepository.java [hunk 31]
```diff
@@ -0,0 +1,19 @@
...
+ * <p/>
+ * Copyright (C) 2021
+ * <p/>
+ * Date: май 20, 2021
+ * @author Sapar
+ */
+@Repository
+public interface IRestaurantRepository {
+
+    List<Restaurant> findAllRestaurants();
```
Comment (AzarenkaA) [inline epam-vote-system-repository/src/main/java/com/epam/vote/repository/IRestaurantRepository.java:18]: comment

File: epam-vote-system-repository/src/main/java/com/epam/vote/repository/IRestaurantRepository.java [hunk 32]
```diff
@@ -0,0 +1,23 @@
...
+ * Copyright (C) 2021
+ * <p/>
+ * Date: май 20, 2021
+ * @author Sapar
+ */
+@Repository
+public interface IRestaurantRepository {
+
+    /**
+     *This method return find all restaurants.
```
Comment (AzarenkaA) [inline epam-vote-system-repository/src/main/java/com/epam/vote/repository/IRestaurantRepository.java:19]: add space after asterisk

File: epam-vote-system-service/src/main/java/com/epam/vote/service/impl/RestaurantService.java [hunk 33]
```diff
@@ -0,0 +1,34 @@
...
+import com.epam.vote.service.IRestaurantService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+/**
+ * Restaurant service.
```
Comment (AzarenkaA) [inline epam-vote-system-service/src/main/java/com/epam/vote/service/impl/RestaurantService.java:14]: I propese to describe as "Implementation of {@link IRestaurantService}"

File: epam-vote-system-service/src/main/java/com/epam/vote/service/impl/RestaurantService.java [hunk 34]
```diff
@@ -0,0 +1,34 @@
...
+public class RestaurantService implements IRestaurantService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(RestaurantService.class);
+
+    @Autowired
+    private IRestaurantRepository restaurantRepository;
+
+    @Override
+    public List<Restaurant> getAllRestaurants() {
+        LOGGER.info("getAllRestaurants is started");
```
Comment (AzarenkaA) [inline epam-vote-system-service/src/main/java/com/epam/vote/service/impl/RestaurantService.java:31]: You should write into log information regarding size of list.

File: epam-vote-system-repository/src/main/java/com/epam/vote/repository/IRestaurantRepository.java [hunk 35]
```diff
@@ -0,0 +1,23 @@
...
+     * @return all restaurants.
```
Comment (AzarenkaA) [inline epam-vote-system-repository/src/main/java/com/epam/vote/repository/IRestaurantRepository.java:20]: list of {@link Restaraunt}'s

File: epam-vote-system-service/src/test/java/com/epam/vote/service/test/RestaurantServiceTest.java [hunk 36]
```diff
@@ -0,0 +1,50 @@
...
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.*;
+
+/**
+ * Test Restaurant service
```
Comment (AzarenkaA) [inline epam-vote-system-service/src/test/java/com/epam/vote/service/test/RestaurantServiceTest.java:18]: comma at the end

**Pull request merged**